### ISSUE TITLE
False positive for no-useless-return

### ISSUE THREAD

**Issue by fasttime** [eslint/eslint#7477]
So I decided to give a try to the new rule `no-useless-return` and I stumbled on what looks to me like _too much of goodness_.

Here is a minimal example to illustrate the issue (actual code is here: https://github.com/fasttime/JScrewIt/blob/b3cce7dbc33628c184a1d39bf827c5b5bb52877e/src/lib/encoder-base.js#L363).

``` js
function returnValidValueOrUndefined(value)
{
    if (!isValidValue(value))
        return; // Unnecessary return statement   no-useless-return
    return value;
}
```

Granted, there are plenty of ways to get rid of the expressionless `return;`, but it seems unfair to call it useless when the next statement would potentially return something different.

**Comment by platinumazure** [comment]
Hmm... I agree, this seems like a bug.

@mysticatea, what do you think? Is this a bug, or working as designed?

**Comment by not-an-aardvark** [comment]
I agree, this is a bug.

**Comment by mysticatea** [comment]
I agree, this is a bug.
I have forgotten to mark as used at `return` statements which have a return value.
